src/Accounts/Authenticators/DeviceCodeAuthenticator.cs

```diff
@@ -0,0 +1,60 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Identity.Client;
+
+namespace Microsoft.Azure.PowerShell.Authenticators
+{
+    public class DeviceCodeAuthenticator : DelegatingAuthenticator
+    {
+        public override Task<IAccessToken> Authenticate(AuthenticationParameters parameters)
+        {
+            var cancellationTokenSource = new CancellationTokenSource();
```

Comment:
Should we be passing in a cancellation token to all of these methods?  The new cmdlets are a cancellation token source, and it would be nice if someone htting ctrl-c could automatically stop authentication.

Reply:
This would mean we'd have to plumb through a cancellation token, but we could default it in the method signatures, so that no cancellation token is necessarily required.